### ISSUE TITLE
Do you have any demo app using it?

### ISSUE THREAD

**Issue by dotnetwise** [zeroasterisk/meteor-cordova-geolocation-background#1]
A live example / perhaps published in appstore / google play would be wonderful.

**Comment by hems** [comment]
@zeroasterisk when running the demo app or trying to roll my own little one, i get an error on XCode

````
2015-03-23 00:28:01.840 app[447:85900] ERROR whitelist rejection: url='http://localhost:3000/sockjs/info?cb=o9tkdoh24a'
````

Any ideas of what might be the issue?

**Comment by zeroasterisk** [comment]
that's prob a new feature of Meteor+Cordova.  You've gotta allow those URLs.
http://docs.meteor.com/#/full/App-accessRule

**Comment by hems** [comment]
@zeroasterisk perhaps you should add instructions on the readme file then ?

**Comment by hems** [comment]
Sorry i'm completely new to meteor, would be something like this ?
````App.accessRule( 'http://mydomain.com', { external: true } ); ````

**Comment by zeroasterisk** [comment]
probably just:

```
App.accessRule( 'http://mydomain.com');
```

Since you want it handled locally (by meteor), not externally (by the phone/other-apps).

Please let me know how it goes... I'll update the docs if this does fix it.

Also, just to be sure you know this, the above App config goes in `<app-root>/mobile-config.js`

http://docs.meteor.com/#/full/mobileconfigjs

**Comment by hems** [comment]
@zeroasterisk funny enough i tried your example now, and it runs on my phone without any errors on xCode.

But when navigating to: https://geolocationbackgroundexample.meteor.com i just see this:

![screen shot 2015-03-30 at 00 52 00](https://cloud.githubusercontent.com/assets/27327/6888483/04c7f6ae-d677-11e4-8bf6-03e691dfadca.png)

My example ( i tried implementing something from "scratch" ) still have the problem with accessing localhost, and for some weird reason i can't get through the initial page.

I'll investigate more on my side, it seems that i have some weird problems.

Does running https://github.com/zeroasterisk/meteor-cordova-geolocation-example in your iphone yields perfect results?

**Comment by jeangui** [comment]
Hello @hems,

I have got the same problem as you and I post an issue on it.

https://github.com/zeroasterisk/meteor-cordova-geolocation-example/issues/3

I think this is related to the data structure that is proposed by the iphone version of the background service. The location attribute is not located at the same level.
Depending on the running context (FG, BG, IOS, Android) the data structure is different. So the meteor server has to be able to transform the data on the fly.

I have been thinking to use matb33/meteor-collection-hooks.

Regards,
@jeangui

**Comment by christocracy** [comment]
Data structure difference:  I did that.

On Monday, March 30, 2015, jeangui <notifications@github.com> wrote:

> Hello @hems <https://github.com/hems>,
>
> I have got the same problem as you and I post an issue on it.
>
> zeroasterisk/meteor-cordova-geolocation-example#3
> <https://github.com/zeroasterisk/meteor-cordova-geolocation-example/issues/3>
>
> I think this is related to the data structure that is proposed by the
> iphone version of the background service. The location attribute is not
> located at the same level.
> Depending on the running context (FG, BG, IOS, Android) the data structure
> is different. So the meteor server has to be able to transform the data on
> the fly.
>
> I have been thinking to use matb33/meteor-collection-hooks.
>
> Regards,
> @jeangui <https://github.com/jeangui>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeroasterisk/meteor-cordova-geolocation-background/issues/1#issuecomment-87591766>
> .
>


-- 
Snet from Gmail Mobile

**Comment by jeangui** [comment]
Hello @christocracy

thanks for your feedback.
I gonna use therefore the meteor package matb33/meteor-collection-hooks.

Best regards

@jeangui

**Comment by zeroasterisk** [comment]
I'd love to see your finished/fixed example.

**Comment by hems** [comment]
@christocracy sorry i got a bit confused, what you mean by "i did that" ?

Did you fix this issue on your package https://github.com/christocracy/cordova-plugin-background-geolocation ? If i'm not mistaken that is the package @zeroasterisk used under the hood here?

**Comment by christocracy** [comment]
I'm responsible for the difference in data-structure from Cordova's
Geolocation plugin (ie:  location.coords)

On Monday, March 30, 2015, henrique matias <notifications@github.com> wrote:

> @christocracy <https://github.com/christocracy> sorry i got a bit
> confused, what you mean by "i did that" ?
>
> Did you fix this issue on your package
> https://github.com/christocracy/cordova-plugin-background-geolocation ?
> If i'm not mistaken that is the package @zeroasterisk
> <https://github.com/zeroasterisk> used under the hood here?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/zeroasterisk/meteor-cordova-geolocation-background/issues/1#issuecomment-87848108>
> .
>


-- 
Snet from Gmail Mobile

**Comment by hems** [comment]
@christocracy and you think would be a good idea to standardize this? any tips on how to get started "fixing" our issue ?

**Comment by zeroasterisk** [comment]
If we have to translate the data FROM this plugin, I think it would have to be on the AJAX receiver on the server... which is not currently implemented as part of this plugin, only as code examples, etc...

If you can give me what JSON we are getting across all possible versions of the plugin (as examples)?

I think making a basic translation method for easy inclusion on the AJAX receiver would be a good idea, and easy.

_(If I'm misunderstanding the issue, please set me straight)_

**Comment by jeangui** [comment]
Hello,

I have not any time to test it but in my mind the best way will be to use the matb33/meteor-collection-hook package with the function .before.insert.
Moreover it will be nice to have a flag on the object sent to the server with the context (FG, BG, android, ios, ...). By this way it will easier to do the transformation on the fly before inserting.

Cheers
@jeangui

**Comment by zeroasterisk** [comment]
I agree - I love the "hooks" approach... but if it's a common problem, I'd like to bundle a common solution with this package, or as an optional but easy secondary package.
